### PR TITLE
feat(init): readable text diff for `lockb`

### LINKED ISSUE #8274: Add `.gitattributes` and `.git/config` in `bun init` to make `git diff` just work for bun.lockb files
Into `.gitattributes`:
```js
*.lockb binary diff=lockb
```

Into `.git/config`:

```
[diff "lockb"]
textconv = bun
binary = true
```

### LINKED ISSUE #8306: `init`: `.gitattributes` & `.gitconfig` to configure diffing `bun.lockb` in a readable format
### What is the problem this feature would solve?

Make `git diff` show readable format of `bun.lockb`.

### What is the feature you are proposing to solve the problem?

Add `.gitattributes` & `.gitconfig` files on `bun init`. ([suggested by @Jarred-Sumner here](https://twitter.com/jarredsumner/status/1748185201224884549)).

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by aboqasem**
### What does this PR do?

Closes #8274

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

- [X] Ran `bun-debug init`
- [X] I modified TypeScript tests and they pass locally (`bun-debug test test/cli/init/init.test.ts`)

### CODE REVIEW

**Comment by aboqasem** [comment]
@Jarred-Sumner apologies I later noticed that you already created an issue. I will modify to include the config in `.git/config` instead of `.gitconfig`.

File: test/cli/init/init.test.ts [hunk 1]
```diff
@@ -72,6 +76,69 @@ test("bun init with piped cli", () => {
...
+  // @ts-expect-error
+  expect(out.signal).toBe(undefined);
+  expect(out.exitCode).toBe(0);
+
+  const gitConfig = fs.readFileSync(path.join(temp, ".git", "config"), "utf8");
+  expect(gitConfig).toEqual(`${gitConfigContent}# Use \`bun\` as the textconv for \`bun.lockb\` files
+[diff "lockb"]
+textconv = bun
+binary = true
+`);
```
Comment (Electroid) [inline test/cli/init/init.test.ts]: Instead of inlining the content, let's just read it from `src/cli`

**Comment by aboqasem** [comment]
Please help review comments under the issue as well: https://github.com/oven-sh/bun/issues/8274#issuecomment-1904462693.

File: src/cli/init_command.zig [hunk 2]
```diff
@@ -370,14 +371,46 @@ pub const InitCommand = struct {
...
+                Output.flush();
+            }
+        }
+
+        if (steps.write_gitconfig) {
+            brk: {
+                if (!exists(".git")) {
+                    destination_dir.makeDirZ(".git") catch break :brk;
+                }
+                const filename = ".git/config";
```
Comment (Electroid) [inline src/cli/init_command.zig:401]: I don't think this will work on Windows.

**Pull request closed without merging**